packages/expo-network/ios/NetworkModule.swift

```diff
@@ -0,0 +1,128 @@
+import ExpoModulesCore
+import SystemConfiguration
+import Network
+
+extension NWInterface.InterfaceType: CaseIterable {
+  public static var allCases: [NWInterface.InterfaceType] = [
+    .other,
+    .wifi,
+    .cellular,
+    .loopback,
+    .wiredEthernet
+  ]
+}
+
+enum NetworkType: CustomStringConvertible {
+  case unknown, wifi, none, cellular
+
+  var description: String {
+    switch self {
+    case .wifi:
+      return "WIFI"
+    case .cellular:
+      return "CELLULAR"
+    case .unknown:
+      return "UNKNOWN"
+    case .none:
+      return "NONE"
+    }
+  }
+}
+
+public final class NetworkModule: Module {
+  private let monitor = NWPathMonitor()
+  private let monitorQueue = DispatchQueue.global(qos: .background)
+  private var type = NetworkType.unknown
+  private var connected = false
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoNetwork")
+
+    OnCreate {
+      startMonitor()
```

Comment:
I'm only wondering if that's fine to monitor the network all the time 🤔
Does the NWPathMonitor, when started, call the path update handler immediately (without waiting for the actual network change)? If so, maybe we could start it only when it's necessary and stop after resolving the promise?

Reply:
I'll give that try. I did it this way because I thought if it's on a background queue, the promise might resolve before it has updated the state. I would think it does run immediately though once it's started.

Reply:
I tried it and starting the monitor in `getNetworkStateAsync()` won't work. The state lags behind so on a second call you have the correct values. Changing the `qos` of the queue to `userInitiated` helps but it's still inconsistent. There is a `currentPath` property on the monitor so I tried just reading that but it just holds the default state if the monitor has not been started, which makes sense. Ideally, we'd want to start the monitor, await the first update, resolve and cancel the monitor.

Reply:
We can avoid the pathUpdateHandler by reading the `currentPath` directly in the function. Probably better for this use case so that the value is always current and not based on queues. Doesn't seem to be anyway around starting the monitor in `OnCreate`